pkg/controller/decorator/controller.go

```diff
@@ -349,6 +350,32 @@ func (c *decoratorController) enqueueParentObjectAfter(obj interface{}, delay ti
 
 func (c *decoratorController) updateParentObject(old, cur interface{}) {
 	// TODO(enisoc): Is there any way to avoid resyncing after our own updates?
+	if parentOld, ok := old.(*unstructured.Unstructured); ok {
+		parentKind := parentOld.GetKind()
+		parentAPIVersion := parentOld.GetAPIVersion()
+		for _, parent := range c.dc.Spec.Resources {
+			resource := c.resources.Get(parent.APIVersion, parent.Resource)
+			if resource == nil {
+				// it's not a resource we care about
+				return
```

Comment:
it should be `continue`

Reply:
you can also add comment that this should not happen, as the resourceMap dicovers all the resources from API server